virttest/vt_console.py

```diff
@@ -73,14 +76,29 @@ def __login(
         self._console.set_linesep(linesep)
         self._console.set_status_test_command(status_test_command)
 
-        is_responsive = False
+        reset_str = "Press any key to stop system reset"
         end_time = time.time() + timeout
         while time.time() < end_time:
-            if self._console.is_responsive():
-                is_responsive = True
-                break
-        if not is_responsive:
-            raise ConsoleNotResponsiveError("Console is not responsive.")
+            self._console.read_nonblocking()
+            latest_output = "\n".join(self._console.get_output().splitlines()[-50:])
+
+            # If system is waiting for any key to stop reset, we'll wait after
+            # timeout(usually 5s) to avoid entering interactive screen
+            if reset_str in latest_output:
+                time.sleep(1)
+                continue
+            else:
```

Comment:
@chloerh Sorry I don't understand what you are achieving here. IIUC, the code at login reads the last lines in the console output and waits as long as the UEFI system reset string is shown? When is this else branch relevant?

Reply:
The logic is:
line 87- 89
first check the latest output, if 'reset' is currently on the screen, we'll redo the loop after sleep(1) aiming to skip the 'reset' screen.

line 90-101
After we cannot find 'reset' in the latest output, it means: 
- the 'reset' has passed
- or it never appeared
if it's the 1st: we will stop the loop and try the login on line 103
if it's the 2nd: we're not sure if the 'reset' is going to appear or not. But we know that if it will appear, it will be at the beginning part of the output, so I set the limit to 500 lines. If no 'reset' after 500 lines, the 'reset' will not appear at all.

Reply:
Thank you very much for explaining. I have run some tests on s390x to confirm there's no negative impact for our systems that don't use that system reset.

May I ask why the current implementation can't be used to wait for the message to disappear?

This looks like quite a specific solution that only applies to UEFI on x86_64 with a specific (qemu?)firmware. I wonder if there could be a general solution that would allow for waiting for any message to disappear first (make the line configurable). However, from https://github.com/avocado-framework/avocado-vt/pull/3948#issuecomment-2242560379 I understand you're considering an alternative solution, so I'll wait for the outcome of that.

Reply:
@smitterl Thanks so much for checking this pr with 390x tests!

Actually, the current implementation was added to workaround the same issue https://github.com/avocado-framework/avocado-vt/commit/2523255ee1f5ff3bbf3d0bcb956d96bcad6691b1 
Unfortunately, it reduced the number of failed cases but not to 0. Therefore we made another workaround to extend internal timeout of login (10s to 20s) which was configured in our ci.

But lately we found some login timeout errors and they could be workaround by time.sleep(x). I have even merged one with passt tests, but I don't think after 2 ways of workaround, adding a 3rd workaround is the best way to deal with this issue. 

> This looks like quite a specific solution that only applies to UEFI on x86_64 with a specific (qemu?)firmware.
About this, I'm not sure about that, but the 'reset' happens quite often after we do xml.sync(). Do you mean that it doesn't happen to other archs? If so, I could check with arm qes to get the info.

Reply:
No, the reset should happen only on x86_64 AFAIK but it could make the code clearer and reusable in case other setups will have a similar timeout, I don't know.

Reply:
I found that `filesystem_device_unprivileged.two_fs..two_guests` test cases reproduce the 'infinite wait for login' with reset message on UEFI guests. But with this patch they pass. I don't really understand how but I will approve this request.

```
JOB ID     : fbb46d18520a1a07a1fc72907fcc27e9f9b63e5b
JOB LOG    : /var/log/avocado/job-results/job-2024-07-24T08.08-fbb46d1/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virtual_devices.filesystem_device_unprivileged.two_fs.with_shm.two_guests.coldplug:  PASS (109.02 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/log/avocado/job-results/job-2024-07-24T08.08-fbb46d1/results.html
JOB TIME   : 110.63 s
```